ml-agents-envs/mlagents/envs/brain_conversion_utils.py

```diff
@@ -0,0 +1,70 @@
+from mlagents.envs.brain import BrainInfo, BrainParameters, CameraResolution
+from mlagents.envs.base_env import BatchedStepResult, AgentGroupSpec
+from mlagents.envs.exception import UnityEnvironmentException
+import numpy as np
+from typing import List, Any
+
+
+def step_result_to_brain_info(
+    step_result: BatchedStepResult,
+    group_spec: AgentGroupSpec,
+    agent_id_prefix: int = None,
+) -> BrainInfo:
+    n_agents = step_result.n_agents()
+    vis_obs_indices = []
+    vec_obs_indices = []
+    for index, observation in enumerate(step_result.obs):
+        if len(observation.shape) == 2:
+            vec_obs_indices.append(index)
+        elif len(observation.shape) == 4:
+            vis_obs_indices.append(index)
+        else:
+            raise UnityEnvironmentException(
+                "Invalid input received from the environment, the observation should "
+                "either be a vector of float or a PNG image"
+            )
+    if len(vec_obs_indices) == 0:
+        vec_obs: List[Any] = []
```

Comment:
I believe in the current code this actually returns an `np.ndarray` of empty `np.ndarray`, of length num_agents. The empty list is causing a crash when we index by agent_index somewhere else in the code.